File: GFG/Minimum number of Coins/Minimum number of Coins.cpp
class Solution{
public:
    vector<int> minPartition(int n)
    {
        // code here
        
          vector<int>v={ 1, 2, 5, 10, 20, 50, 100, 200, 500, 2000 };

        vector<int>ans;

        int p,q;

        while(n!=0){

            for(int i=9;i>=0;i--){

                if(n/v[i]!=0){

                    int x=n/v[i];

                    while(x--){

                        n-=v[i];

                        ans.push_back(v[i]);

                    }    

                }

            }

        }

        return ans;
    }
};

File: GFG/Minimum number of Coins/Minimum number of Coins.java
class Solution{
    static List<Integer> minPartition(int N)
    {
        // code here
        int[]a=new int[]{1,2,5,10,20,50,100,200,500,2000};
        List<Integer> ans=new ArrayList<>();
        int i=9;
        while(i>=0){
            if(a[i]>N){
                i--;
            }else{
                while(N>=a[i]){
                    ans.add(a[i]);
                    N-=a[i];
                }
            }
        }
        return ans;
    }
}
